src/_P037_MQTTImport.ino

```diff
@@ -456,7 +456,7 @@ boolean Plugin_037(byte function, struct EventStruct *event, String& string)
                   RuleEvent += event->String1;
                   RuleEvent += '=';
                   RuleEvent += unparsedPayload;
-                  eventQueue.add(RuleEvent);
+                  eventQueue.addMove(std::move(RuleEvent));
```

Comment:
Try also to reserve the expected length of the RuleEvent.
After the move, its content is empty, so the String copied to the eventQueue has the same properties, including the `capacity`, which may have been overdimensioned if just resized based on `+=`.
Not entirely sure how Arduino's `String` class does the resize, whether it is always the next multiple of 16 or increasing step size based on the current capacity (most std::string implementations increment step size when allocating more memory)